src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1094,6 +1094,7 @@ internal async Task UpdateDetailPaneAsync(CancellationToken cancellationToken)
                 EmitSearchSelectionTelemetry(selectedItem);
 
                 await _detailModel.SetCurrentPackageAsync(selectedItem, _topPanel.Filter, () => _packageList.SelectedItem);
+                Model.UIController.SelectedPackageId = selectedItem.Id;
```

Comment:
The main PMUI control knows that a package was changed, so I think the (parent) PMUI model should also store the selected package ID value. 
Previously, we were only passing it to the Details Pane model.